lib/level_db.rb

```diff
@@ -113,6 +113,24 @@ def each(options={}, &block)
     def snapshot
       Snapshot.new(@db)
     end
+
+    def compact_range(options={})
+      from = options[:from]
+      to = options[:to]
+      @db.compact_range(encode_key(from), encode_key(to))
+    end
+
+    def full_compaction
+      sstables = @db.get_property(SSTABLES_PROPERTY)
+      pre_compaction_sstables = nil
+      until pre_compaction_sstables == sstables
+        pre_compaction_sstables = sstables
+        compact_range
+        sstables = @db.get_property(SSTABLES_PROPERTY)
+      end
```

Comment:
Thank you for noticing, I totally forgot about this. My original intention was to add a maximum iterations options which would terminate the compactions regardless of convergence. I'll revise it.

And also, there are other properties called `leveldb.num-files-at-level<N>` where `N` is an integer. Do you think it'd be nicer/more clear to have a termination condition based on the number of files? It is a bit cumbersome to count them all, since there's no way of knowing how many levels there are, so you'd have to do something like
```ruby
count = i = 0
while (value = @db.get_property("leveldb.num-files-at-level#{i}"))
  count += value.to_i
  i += 1
end
count
```
My main concern about the current solution is that I'm not sure about the contents of the `leveldb.sstables` property, and specifically its sorting. It's just a multi-line string that describes all sstables, but I don't know if its order is deterministic. I'm guessing that it is, but I don't know. And it feels like an unstable termination condition to make a string comparison like that when the format isn't well documented.